modules/generate/generate.go

```diff
@@ -34,6 +35,64 @@ func GetRandomString(n int) (string, error) {
 	return string(buffer), nil
 }
 
+// Password generates a random password
+func Password(n int) (string, error) {
+	const lower = "abcdefghijklmnopqrstuvwxyz"
+	const upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	const digit = "0123456789"
+	const spec = "_-"
+	var h, m int
+	if n < 3 {
+		return "", errors.New("Error generate random string")
+	}
+
+	h = (n - 2) / 2
+	m = (n) % 2
+	buffer := make([]byte, n)
+	maxLower := big.NewInt(int64(len(lower)))
+	maxUpper := big.NewInt(int64(len(upper)))
+	maxDigit := big.NewInt(int64(len(digit)))
+	maxSpec := big.NewInt(int64(len(spec)))
+
+	for i := 0; i < h; i++ {
```

Comment:
I don't think this is the proper way to increase the security. To give a simple example, let's say upper is "ABC" and lower is "abc". If I set the rules for `upper, lower`, length=6, it must not mean that I will have 50% uppercase characters and 50% lowercase, like: `AACcba`, `CBBacb`, etc. That's 3^3+3^3 = 54 possible passwords. The way this should be handled is a 6 character string where any character can come from the whole set `ABCabc`, like: `AcbABC`, `AAAcABC`, 'acabbba`, etc. That's 6^6 = 46656 possibilities.
Sorry I didn't spot this sooner.

Reply:
The way I'd do this is to loop through random strings from the available character classes and run it by `util.CheckPasswordComplexity()` to verify it contains at least one character from each required class. If the string doesn't comply with the rule, try again until one does.

Reply:
Ok,  you are right

Reply:
@guillep2k 
Should password generation take the current password complexity settings from the configuration file?

Reply:
I don't understand the question. It should follow the same rules as required for users; otherwise the users will be confused.